barman/backup_executor.py

```diff
@@ -1593,6 +1601,60 @@ def stop_backup(self, backup_info):
             # Skip switching XLOG if a standby server
             pass
 
+    @contextmanager
+    def _get_tar_fileobj(self, backup_info):
+        compressions = {"gzip": "gz", "lz4": "lz4", "zstd": "zst"}
+        try:
+            tar_filename = "base.tar.%s" % compressions[backup_info.compression]
+        except KeyError:
+            # Any attempt to use an unsupported compression should be caught in
+            # the CLI argument handling so if we get here it is almost certainly
+            # a bug
+            raise BackupException(
+                "Cannot determine file suffix for %s compressed pg_basebackup"
+                % backup_info.compression
+            )
+
+        tar_path = os.path.join(backup_info.get_data_directory(), tar_filename)
+
+        if backup_info.compression == "gzip":
+            yield open(tar_path, "rb")
+        elif backup_info.compression == "lz4":
+            import lz4.frame
+
+            yield lz4.frame.open(tar_path, mode="rb")
+        elif backup_info.compression == "zstd":
+            import pyzstd
+
+            yield pyzstd.open(tar_path, "rb")
+
+    def _read_compressed_backup_label(self, backup_info):
```

Comment:
Reading a `backup_label` from a backup compressed by pg_basebackup does seem like the job of the `PostgresBackupStrategy` so I think it is reasonable for this code to live here.